src/components/field.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { Fragment } from 'react';
+import { ContextApi, ContextForm } from './form-context';
+import { ValueType } from './validator';
+
+export interface FieldProps {
+  name: string;
+  className?: string;
+  hideError?: boolean;
+  disabled?: boolean;
+}
+
+export interface FieldPassedProps {
+  name: string;
+  disabled: boolean;
+  className: string;
+  getValue: () => ValueType;
+  setValue: (value: ValueType, callback?: () => void) => void;
+  setTouched: (callback?: () => void) => void;
+}
+
+type Diff<T, U> = Pick<T, Exclude<keyof T, keyof U>>;
+type Optional<T, K extends keyof T> = Omit<T, K> & Partial<Pick<T, K>>;
+
+const Field = <P extends FieldProps>(
+  PassedComponent: React.ComponentType<P>,
+  passedProps: Partial<Diff<P, FieldPassedProps> & FieldProps> = {},
+) => {
+  const FieldComponent: React.FC<Diff<Optional<P, keyof typeof passedProps>, FieldPassedProps> & FieldProps> = (
+    fieldProps, // Exclude<P, FieldPassedProps> & FieldPassedProps
+  ) => (
+    <ContextApi.Consumer>
+      {api => (
+        <ContextForm.Consumer>
+          {() => {
+            const { name, className, hideError, disabled, ...props } = { ...fieldProps, ...passedProps };
+            if (!api) {
+              console.error(`Could not found Form API. Make sure <${PassedComponent.displayName}/> is in the <Form/>.`);
+              return null;
+            }
+            const errors = api.getErrors(name);
+            const classNames = className ? [className] : [];
+            const errorClassName = api.getInvalidClass();
+            if (errors && errors.length > 0 && errorClassName) {
```

Comment:
in such case we can use [optional chaining](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining)
```
if (errors?.length > 0 && errorClassName) {
```